src/evolution.jl

```diff
@@ -215,13 +253,22 @@ function add_branch_and(
     if idx == nothing
         possible_ids = Vector{Int}()
         inside_and = false
+        forbidden_iter = 0
         for (i, e) in enumerate(chromo)
-            if isletter(e) && !inside_and
+            if isletter(e) && !inside_and && forbidden_iter == 0
                 push!(possible_ids, i)
             elseif e == '['
                 inside_and = true
             elseif e == '}'
                 inside_and = false
+            elseif e == '('
+                l_bracket, pipe, r_bracket, _ = find_brackets(chromo, i)
+                if pipe == nothing
+                    forbidden_iter = r_bracket
+                end
+            end
```

Comment:
czy w każdym elseifie nie chcemy sprawdzać czy forbidden_iter jest 0?
Myślę o case:
ab(asb(fas|fda)dfa|dasf)
czy tutaj po zamknięciu zagnieżdzonego ora nie będzie już forbidden_iter, czy to chcemy?

Reply:
a w przykładzie ab(ab(ab)+ab)+ ?

Reply:
Nie może być zagnieżdzonych takich nawiasów. To jest założenie, na które się kiedyś zgodziliśmy, ponieważ nie chcemy loopów w loopach.

Reply:
okej, to git